src/transformation/visitors/loops/do-while.ts

```diff
@@ -1,23 +1,51 @@
 import * as ts from "typescript";
 import * as lua from "../../../LuaAST";
 import { FunctionVisitor } from "../../context";
-import { transformLoopBody } from "./utils";
+import { invertCondition, transformLoopBody } from "./utils";
 
-export const transformWhileStatement: FunctionVisitor<ts.WhileStatement> = (statement, context) =>
-    lua.createWhileStatement(
-        lua.createBlock(transformLoopBody(context, statement)),
-        context.transformExpression(statement.expression),
-        statement
-    );
+export const transformWhileStatement: FunctionVisitor<ts.WhileStatement> = (statement, context) => {
+    const body = transformLoopBody(context, statement);
+
+    context.pushPrecedingStatements();
+    let condition = context.transformExpression(statement.expression);
+    const precedingStatements = context.popPrecedingStatements();
+
+    // Change from 'while condition' to 'while true - if not condition then break'
```

Comment:
I guess the reason for this is to re-evaluate the preceding statements every iteration? Could you add that as a comment?